File: Go/Decorator Pattern/milk.go
package main

type Milk struct {
	BeverageDecorator
}

func NewMilk(b Beverage) Beverage {
	return Milk{BeverageDecorator{Beverage: b}}
}

func (m Milk) Cost() float64 {
	return m.Beverage.Cost() + 0.20
}

func (m Milk) Description() string {
	return m.Beverage.Description() + ", Milk"
}

File: Design Patterns/JS/Decorator Pattern/milk.js
import BeverageDecorator from "./beverageDecorator.js";

class Milk extends BeverageDecorator {
  cost() {
    return super.cost() + 0.2;
  }

  description() {
    return super.description() + ", Milk";
  }
}

export default Milk;
